src/libbson/src/bson/bson-clock.c

```diff
@@ -126,18 +126,8 @@ bson_get_monotonic_time (void)
    clock_gettime (CLOCK_MONOTONIC, &ts);
    return (((int64_t) ts.tv_sec * 1000000) + (ts.tv_nsec / 1000));
 #elif defined(__APPLE__)
-   static mach_timebase_info_data_t info = {0};
-   static double ratio = 0.0;
-
-   if (!info.denom) {
-      /* the value from mach_absolute_time () * info.numer / info.denom
-       * is in nano seconds. So we have to divid by 1000.0 to get micro
-       * seconds*/
-      mach_timebase_info (&info);
-      ratio = (double) info.numer / (double) info.denom / 1000.0;
-   }
-
-   return mach_absolute_time () * ratio;
```

Comment:
>  by delegating the required intermediate floating-point arithmetic to the implementation of clock_gettime_nsec_np(CLOCK_UPTIME_RAW).

Does this suggest multiplying by `ratio` could be the cause of a decrease in return values? If so, that surprises me.

Reply:
I suspect there may be some floating-point shenanigans due to the potentially large magnitudes of values being converted to and computed in the floating-point domain (coarseness? rounding?) before being converted back into the integral domain.